uqcsbot/api.py

```diff
@@ -56,16 +63,27 @@ def __call__(self, **kwargs) -> dict:
 
         Attempts to retry the API call if rate-limited.
         """
-        fn = partial(self._client.api_call, self._method, **kwargs)
+        def get_fn(call_type):
+            client = getattr(self, f'_{call_type}_client')
+            method = getattr(client, self._method.replace('.', '_'))
+            return partial(method, **kwargs)
         retry_count = 0
+        call_type = _CLIENT_METHOD_REGISTRY.get(self._method, 'bot')
         while retry_count < 5:
-            result = fn()
+            try:
+                result = get_fn(call_type)()
+            except slack.errors.SlackApiError as e:
+                result = e.response
             if not result['ok'] and result['error'] == 'ratelimited':
                 retry_after_secs = int(result['headers']['Retry-After'])
                 LOGGER.info(f'Rate limited, retrying in {retry_after_secs} seconds')
                 time.sleep(retry_after_secs)
                 retry_count += 1
+            elif not result['ok'] and result['error'] == 'not_allowed_token_type':
+                call_type = {'bot': 'user', 'user': 'bot'}[call_type]
+                retry_count += 1
```

Comment:
So this could switch back and forth between bot and user accounts 5 times? It won't break anything, but it doesn't seem ideal. Can we fail out of this if we've already tried both accounts?

Reply:
Yeah this was just laziness lel